### ISSUE TITLE
Delayed tooltip on button does not appear on first hover after re-enabling

### ISSUE THREAD

**Issue by wrummler** [angular-ui/bootstrap#4204]
I'd been watching #3611, and when it resolved I found this behavior where a delayed tooltip on a button would not appear when first mousing over the button after the button had been re-enabled after being disabled.

Here is a plunker example with repro steps in the description:

http://plnkr.co/edit/AvVRQ5h7K4VNGO70jMrF?p=preview

Here is the example template (self-contained):

    <!DOCTYPE html>
    <html ng-app="ExampleApp">

    <head>
      <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css" />
      <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.3/angular.min.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-bootstrap/0.13.3/ui-bootstrap-tpls.min.js"></script>
      <script>
        angular.module('ExampleApp', ['ui.bootstrap']);
      </script>
    </head>

    <body>
      <h1>Hello Plunker!</h1>
      <div ng-show="show">
        <span style="font-size: 400%">HELLO USER</span>
        <button class="btn btn-xs btn-default" ng-click="show = false">Hide</button>
      </div>
      <button class="btn btn-xs btn-default" ng-click="show = true" tooltip="Click me!" tooltip-popup-delay="1000" ng-disabled="show">Retaliatorily Salutate</button>
    </body>

    </html>

Here are the repro steps:

1. Click button before tooltip shows.
2. Click "Hide" button.
3. Hover over first button. Tooltip never appears.
4. Mouse out from first button, then mouse back in. Tooltip will appear.

**Comment by wesleycho** [comment]
This appears to be reproduced if an element is disabled - this is because jqLite does not recognize mouseenter events on disabled elements in Angular, as can be seen [here](http://plnkr.co/edit/I9nfaufIcmLveisBq9GG?p=preview). Doing a bit of research into this, this appears to have been incorrect behavior in jQuery, and it was replicated likely to match the original behavior in jQuery - jQuery fixed this behavior it looks like from [this ticket](http://bugs.jquery.com/ticket/11382).

**Comment by wesleycho** [comment]
Hm, looking into this deeper, it appears that jQuery too also does not log the event, nor does the regular browser, as can be seen [here](http://plnkr.co/edit/oBdTBWAxGMqEC42CuoqB?p=preview)

I am inclined to say that this is not a bug unfortunately. One way around this is to take advantage of the new programmatic control of the tooltip with the `tooltip-is-open` attribute, but that is only currently available on `master` unfortunately.

**Comment by wrummler** [comment]
@wesleycho Note that the original example reproduces the behavior when the button is enabled, not disabled.

**Comment by wesleycho** [comment]
Just noting a verification that the event listeners behave as expected independent of tooltips [here](http://plnkr.co/edit/3TU7oVr4GZLlHQuCMpP4?p=preview) - this requires investigation into tooltip behavior.

**Comment by wesleycho** [comment]
I went through and debugged a little with a forked version of the Plunker [here](http://plnkr.co/edit/xuR9zYuM2KvEN5zXp4QK?p=preview), and this actually is a garbage collection bug with the cancellation of the timeout.

The fix is very simple it turns out.

**Comment by wrummler** [comment]
Awesome! :+1: Thanks @wesleycho
